cluster/src/main/scala/io/snappydata/gemxd/SparkSQLPrepareImpl.scala

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) 2016 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package io.snappydata.gemxd
+
+import java.io.DataOutput
+
+import scala.collection.mutable
+
+import com.gemstone.gemfire.DataSerializer
+import com.gemstone.gemfire.internal.shared.Version
+import com.pivotal.gemfirexd.internal.engine.Misc
+import com.pivotal.gemfirexd.internal.engine.distributed.message.LeadNodeExecutorMsg
+import com.pivotal.gemfirexd.internal.engine.distributed.{GfxdHeapDataOutputStream, SnappyResultHolder}
+import com.pivotal.gemfirexd.internal.iapi.types.{DataType => _, _}
+import com.pivotal.gemfirexd.internal.shared.common.StoredFormatIds
+import com.pivotal.gemfirexd.internal.snappy.{LeadNodeExecutionContext, SparkSQLExecute}
+
+import org.apache.spark.Logging
+import org.apache.spark.sql.{Row, SnappySession}
+import org.apache.spark.sql.catalyst.expressions.{BinaryComparison, CaseWhen, Cast, Exists, Expression, Like, ListQuery, ParamLiteral, PredicateSubquery, ScalarSubquery, SubqueryExpression}
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.internal.SnappySessionState
+import org.apache.spark.sql.types._
+import org.apache.spark.util.SnappyUtils
+
+
+class SparkSQLPrepareImpl(val sql: String,
```

Comment:
This was done to avoid following line in SparkSQLExecuteImpl:
  private[this] val df = session.sql(sql)

Care has been taken to reuse some other methods of SparkSQLExecuteImpl by moving them to an Object.